### DISCUSSION TITLE
DJ Connecté

### DISCUSSION THREAD

**Discussion by Moustick2fr** [AzuraCast/AzuraCast#5931] [General]
Bonjour, je recherche un script qui me permet d'extraire le nom du dj connecté sur azuracast mais je ne trouve pas. 

Est ce que quelqu'un aura une idée ou un script pour l'extraire de playingnow.

Merci d'avance.

Désolé je ne parle que français

**Comment by Mozitux** [comment]
Hello, I think that you will potentially have more answers to your question if you ask this one in English. :)

**Comment by Moustick2fr** [comment]
Hello, I'm looking for a script that allows me to extract the name of the dj connected to azuracast but I can't find it.

Does anyone have any idea or script to extract it from playingnow.

Thank you

**Reply by Mozitux** [reply]
Bonsoir, tu cherches à l'extraire pour quel usage ? (L'afficher sur une page web ? Sous forme .json ?)

Cette information nous permettra de te donner la marche à suivre.
####
Hi, what are you trying to extract it for?  (Display it on a web page? As .json?)

 This information will allow us to give you the best procedure to make what you wish.

**Reply by Moustick2fr** [reply]
bonsoir, je cherche à extraire juste le nom du dj connecté dans le but de créer un système de bannière. 
Je m'explique dès que le dj est connecté son nom sera associé à un fichier jpeg et s'affiche automatiquement sur le site web, dès qu'Il se déconnecte une bannière par défaut s'affichera automatiquement.
Merci pour votre aide.

**Reply by VParvulus** [reply]
Il y a quelques exemples sur la façon de travailler avec l'api now playing dans les docs : https://docs.azuracast.com/en/developers/apis/now-playing-data

Vous aurez probablement besoin de quelque chose comme ceci (en supposant que jQuery est disponible) :

---

There are some examples on how to work with the now playing api in the docs: https://docs.azuracast.com/en/developers/apis/now-playing-data

You'll probably need something like this (assuming jQuery is available):

```js
var nowPlayingTimeout;

function loadNowPlaying() {
    $.ajax({
        cache: false,
        dataType: "json",
        url: 'http://your-azuracast-site.example.com/api/nowplaying/station_shortcode',
        success: function(np) {
            // This will print the streamer name in the javascript console
            //-----------
            // Ceci imprimera le nom du streamer dans la console javascript.
            console.log(np.live.streamer_name);

            nowPlayingTimeout = setTimeout(loadNowPlaying, 15000);
        }
    }).fail(function() {
        nowPlayingTimeout = setTimeout(loadNowPlaying, 30000);
    });
}

$(function() {
    loadNowPlaying();
});
```

**Reply by Moustick2fr** [reply]
merci je vais essayer avec ça, je reviens vers toi pour des nouvelles

**Reply by Moustick2fr** [reply]
bon je pense que mes compétences sont plus limitées que je le pensais car je n'y arrive pas dommage
